Makefile

```diff
@@ -67,15 +67,16 @@ generate: export PATH := $(go_bin):$(PATH)
 generate: $(protoc_bin) ## Generate code with controller-gen and protobuf.
 	go version
 	rm -rf apis/generated
-	go run sigs.k8s.io/controller-tools/cmd/controller-gen paths=./apis/... object crd:crdVersions=v1,allowDangerousTypes=true output:artifacts:config=./apis/generated
-	go run sigs.k8s.io/controller-tools/cmd/controller-gen rbac:roleName=appcat paths="{./apis/...,./pkg/apiserver/...}" output:artifacts:config=config/apiserver
+	go run sigs.k8s.io/controller-tools/cmd/controller-gen@v0.15.0 paths=./apis/... object crd:crdVersions=v1,allowDangerousTypes=true output:artifacts:config=./apis/generated
+	go run sigs.k8s.io/controller-tools/cmd/controller-gen@v0.15.0 rbac:roleName=appcat paths="{./apis/...,./pkg/apiserver/...}" output:artifacts:config=config/apiserver
 	go generate ./...
-	go run k8s.io/code-generator/cmd/go-to-protobuf \
+	go run k8s.io/code-generator/cmd/go-to-protobuf@v0.26.3 \
```

Comment:
I would suggest to manage versions via go.mod. Just move these to the `tools.go` file

Reply:
`sigs.k8s.io/controller-runtime`  - this packed throws many errors after I do this upgrade and upgrading it opens new hellhole of errors. This is why I choose such an approach

Reply:
I think the issue might be with Go version itself, it's pretty new and a lot of packages must introduce compability with it

Reply:
I didn't realise that updating the controller-tools library will trigger updating all the other dependencies. We already know that the current api-server is not maintained and any attempts to update dependencies will make things broken. So jsut keep what you have here for now. Bring this topic into the team discussion.